x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/FieldExtractorTestCase.java

```diff
@@ -1715,7 +1724,23 @@ private static Map<String, Object> columnInfo(String name, String type) {
     }
 
     private static Map<String, Object> unsupportedColumnInfo(String name, String... originalTypes) {
-        return Map.of("name", name, "type", "unsupported", "original_types", List.of(originalTypes));
+        DataType suggested = DataType.suggestedCast(
+            List.of(originalTypes).stream().map(DataType::fromTypeName).filter(Objects::nonNull).collect(Collectors.toSet())
+        );
+        if (suggested == null) {
+            return Map.of("name", name, "type", "unsupported", "original_types", List.of(originalTypes));
+        } else {
+            return Map.of(
```

Comment:
Could you use `Map.ofEntries` here? It makes the formatter much more likely to put each key/value pair on one line.